packages/frontend-2/components/server-management/DeleteProjectDialog.vue

```diff
@@ -0,0 +1,48 @@
+<template>
+  <LayoutDialog v-model:open="isOpen" max-width="sm" :title="title" :buttons="buttons">
+    <div class="flex flex-col gap-6">
+      <p>
+        Are you sure you want to
+        <strong>permanently delete</strong>
+        the selected project?
+      </p>
+      <div v-if="project">
+        <strong>
+          {{ isProject(project) ? project.name : '' }}
+        </strong>
+        <p>
+          {{ isProject(project) ? project.models.totalCount : '' }} models,
+          {{ isProject(project) ? project.versions.totalCount : '' }} versions,
+        </p>
+      </div>
+      <p>
+        This action
+        <strong>cannot</strong>
+        be undone.
+      </p>
+    </div>
+  </LayoutDialog>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import { LayoutDialog } from '@speckle/ui-components'
+import { ItemType } from '~~/lib/server-management/helpers/types'
+import { isProject } from '~~/lib/server-management/helpers/utils'
+
+const emit = defineEmits<{
+  (e: 'update:open', val: boolean): void
+}>()
+
+const props = defineProps<{
+  title: string
+  open: boolean
+  project: ItemType | null
```

Comment:
Why is this `ItemType` (Project or User or Invite)? It's a "delete project" dialog, passing in invites and/or users is not a valid use case. Additionally, because of this you need the extraneous `isProject` checks in the template